pkg/apis/policy/v1alpha1/propagation_types.go

```diff
@@ -164,6 +168,60 @@ type ClusterAffinity struct {
 	ExcludeClusters []string `json:"exclude,omitempty"`
 }
 
+// ReplicaSchedulingStrategy represents the assignment strategy of replicas. each ReplicaSchedulingStrategy can not be co-exist, validate through webhook
+type ReplicaSchedulingStrategy struct {
+	// DuplicatePreferences means that each cluster will have the same replicase in workload
+	// +optional
+	DuplicatePreferences bool `json:"duplicatePreferences,omitempty"`
+
+	//WeightPreferences describes weight for each cluster or for each group of cluster
+	// +optional
+	WeightPreferences *ClusterPreferences `json:"weightPreferences,omitempty"`
+
+	// ResourcesPreferences represents a pattern to assign replicas considering the resources in need
+	// default ReplicaSchedulingStrategy. if there is no strategy be set explicitly
+	// +optional
+	ResourcesPreferences *AggregationPreferences `json:"resourcesPreferences,omitempty"`
+}
+
+// ClusterPreferences describes weight for each cluster or for each group of cluster.
+type ClusterPreferences struct {
+	// StaticWeightList defines the static cluster weight.
+	// +required
+	StaticWeightList []StaticClusterWeight `json:"staticWeightList"`
+}
+
+// StaticClusterWeight defines the static cluster weight.
+type StaticClusterWeight struct {
+	// TargetCluster describes the filter to select clusters.
+	// +required
+	TargetCluster ClusterAffinity `json:"targetCluster"`
+
+	// Weight expressing the preference to the cluster(s) specified by 'TargetCluster'.
+	// +required
+	Weight int64 `json:"weight"`
+}
+
+// AggregatedType is the type to define AggregationPattern
+type AggregatedType string
+
+// Available fields for spreading are: cluster, region, zone, and provider.
```

Comment:
Error in comments. Available fields for spreading are: cluster, region, zone, and provider?

Reply:
My mistake and I will modify it.